### DISCUSSION TITLE
Flask API for TM1 fetch and jsonify data timed out exception

### DISCUSSION THREAD

**Discussion by Rvee93** [cubewise-code/tm1py#968] [Q&A]
I am trying to write an API which will fetch the data from cube name and a cube view I will mention in the request. 

Currently it's working for me with views having lesser data but when the data exceeds like 10,000 to 500,000 cells to be fetched is showing me timeout exception while fetching the data. Can you please guide what should I do to fix it.

**Comment by MariusWirtz** [comment]
Hi @Rvee93,

What is the exact error message that you get?

Are you on PAoC?
Please try to use `async_requests_mode`:

``` python
from TM1py import TM1Service

tm1_local_params = {
    'base_url': 'https://localhost:12354',
    'user': 'admin',
    'password': 'apple',
    "async_requests_mode": True
}

with TM1Service(**tm1_local_params) as tm1:
    print(tm1.server.get_server_name())
```
